compiler/src/main/java/bullet/impl/ComponentProcessingStep.java

```diff
@@ -114,35 +124,32 @@ private void generateObjectGraph(TypeElement element) {
           break;
         case SIMPLE_MEMBERS_INJECTION:
         case MEMBERS_INJECTOR:
-          membersInjectionMethods.add(methodDescriptor);
+          memberInjectorsTypeGraph.addVertex(methodDescriptor);
+          memberInjectionMethodsTypeDescriptorMap.put(methodDescriptor.type(), methodDescriptor);
+
+          for (ComponentMethodDescriptor pendingChild : potentialPendingEdges.get(methodDescriptor.type())) {
+            memberInjectorsTypeGraph.addEdge(pendingChild, methodDescriptor);
+          }
+
+          DeclaredType declaredType = (DeclaredType) methodDescriptor.type();
+          TypeElement typeElement = (TypeElement) declaredType.asElement();
+
+          List<TypeMirror> parents = new ArrayList<>(typeElement.getInterfaces());
+          parents.add(typeElement.getSuperclass());
+          for (TypeMirror parent : parents) {
+            ComponentMethodDescriptor parentDescriptor = memberInjectionMethodsTypeDescriptorMap.get(parent);
+            if (parentDescriptor == null) {
+              potentialPendingEdges.put(parent, methodDescriptor);
```

Comment:
We could do this, which is O(n^2), or maybe better just add the whole reachable type hierarchy (like a transitive closure). I mean, after the loop start processing all the pending edges, which contain parent nodes which the component has no member injection of. We then add the parents of the parents, and in this way we end up traversing the graph. This has the drawback of adding unnecessary if conditions to the cascade, but it won't affect anything. However this is cheaper, as it's O(V + E) in a DAG, which in the worst case it could be O(n^2), but in all cases is more efficient.